### PR TITLE
I2317 Add 'metadata' drop down box

### PR DESCRIPTION
**Pull request by JamesThompson1729**
This PR adds a drop-down panel that contains human readable meta-data describing the data used to produce the plot. This came about a request from the funders that they wanted something they cut and paste to remind them what the plot represents.

### CODE REVIEW

File: src/MetaDataDisplay.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+export function MetaDataDisplay(chartOptions: CustomChartOptions,
+	                            human: boolean): string {
+	const maxPlot: number = chartOptions.maxPlot;
+
+	const pltType: string = "Impact / Time series";
+	const cumul: string = "cum:TRUE / cum:FALSE"
+
+	if (human) { // produce human readable metadata
+		if (chartOptions.plotType === "Impact") {
+			const metaStr : string = "This plot shows the " +
```
Comment (EmmaLRussell) [inline src/MetaDataDisplay.ts]: It looks like the first sentence for both impact and time series ("This plot shows the..") should be the same, so could just have the code to calculate that once and then append the second sentence depending on plot type. The string construction could be more readable using template strings.

File: src/MetaDataDisplay.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+									 chartOptions.metric.replace("_", " ") + " data for:\n" +
+			                         prettyActivities(chartOptions.activityTypes) + ";\n" + 
+			                         prettyCountries(chartOptions.selectedCountries, 4) + ";\n" +
+			                         prettyVaccines(chartOptions.selectedVaccines, 4) + ";\n" +
+			                         prettyTouchstones(chartOptions.selectedTouchstones, 2) + ";\n" +
+			                         prettyYears(chartOptions.yearLow, chartOptions.yearHigh) + ".\n" +
+			                         "The data is divide up by " +
```
Comment (EmmaLRussell) [inline src/MetaDataDisplay.ts]: Typo at the end of 'divide'? Should be 'divided'?

File: src/MetaDataDisplay.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			                         prettyVaccines(chartOptions.selectedVaccines, 4) + ";\n" +
+			                         prettyTouchstones(chartOptions.selectedTouchstones, 2) + ";\n" +
+			                         prettyYears(chartOptions.yearLow, chartOptions.yearHigh) + ".\n" +
+			                         "Each line represents a " + chartOptions.disagg +
+			                         ".\nPlot produced at " + prettyDataAndTime();
+			return metaStr;
+		} else {
+			return "ERROR!";
+		}
+	}
```
Comment (EmmaLRussell) [inline src/MetaDataDisplay.ts]: No explicit return value if not human?

File: src/index.html [hunk 4]
```diff
@@ -370,17 +373,21 @@
...
-                    <button class="btn btn-success" data-bind="click: renderMap">Redraw Plot</button>
-                </div>
+        <div class="card" style="width: 50rem;">
+            <div class="card-header">
+                <a class="nav-link nav-parent"
+                   data-toggle="collapse"
+                   href="#showHideMeta"
+                   aria-expanded="false"
+                   aria-controls="showHideMeta"
+                   >Show / Hide metadata</a>
```
Comment (EmmaLRussell) [inline src/index.html]: You could switch this text depending on what its current state is - either 'Show metadata' or 'Hide metadata'. Will 'metadata' be meaningful for all users? Would 'description' make more sense?

File: src/data_vis.ts [hunk 5]
```diff
@@ -192,6 +194,10 @@ class DataVisModel {
         };
     }, this).extend({rateLimit: 250});
 
+    private metaData = ko.computed<string>(() => {
```
Comment (EmmaLRussell) [inline src/data_vis.ts]: This updates nicely when I change the outcome for the plot (deaths to dalys) but when I reduce the number of bars so I only see Pakistan, Ethiopia and Nigeria in the plot, the metadata still says 'Countries: India, Pakistan, Nigeria, Ethiopia'. Is that expected?
screenshot from 2019-03-08 10-26-10

Comment (JamesThompson1729) [reply src/data_vis.ts]: This is difficult; the user has set the sidebar filters to include four countries. but then asked to only only show the largest three.

it's not clear if we should list all of the countries that were selected but that only the largest three are shown. Or only list the largest three.

It is easier to link the metadata to what the user asked for (first option) than to the filtered data  (second option). I'll implement the former, then if that still looks wrong I'll think about the latter.

**Pull request merged**